### ISSUE TITLE
Tartube portable is not portable (path problem)

### ISSUE THREAD

**Issue by zjarhd** [axcore/tartube#494]
### What operating system are you using?
> Window10

### What version of Tartube are you using?
> 2.4.221

### If your bug report is "I can't download this video", please provide a link to the video This will save a lot of time!
> test with https://www.youtube.com/@TheHDRChannel/videos

when tartube folder is moved, it opens
![1](https://user-images.githubusercontent.com/37343691/212462976-699bd62b-0402-4fa2-9272-950c6d3643c1.PNG)

if you set path to new path( E:\newfolder\tartube-2.4.221-64bit-portable\\!backup ) it works fine.
but if you click 'D/L & watch' or 'Check marked items' or 'Download marked items', it doesn't work and it opens
![2](https://user-images.githubusercontent.com/37343691/212463310-e6d9f524-96a7-415f-bfef-39479c120d46.PNG)

```
Administrator@DESKTOP-HN6A9H2 MINGW64 /home/user/tartube
# python3 tartube/tartube
E:/newfolder/tartube-2.4.221-64bit-portable/mingw64/lib/python3.10/site-packages/gi/overrides/Gio.py
:42: Warning: unable to set property 'strikethrough-set' of type 'gboolean' from value of type 'gcha
rarray'
  return Gio.Application.run(self, *args, **kwargs)
E:/newfolder/tartube-2.4.221-64bit-portable/mingw64/lib/python3.10/site-packages/gi/overrides/Gio.py
:42: Warning: unable to set property 'underline-set' of type 'gboolean' from value of type 'gchararr
ay'
  return Gio.Application.run(self, *args, **kwargs)
E:/newfolder/tartube-2.4.221-64bit-portable/mingw64/lib/python3.10/site-packages/gi/overrides/Gio.py
:42: Warning: unable to set property 'weight-set' of type 'gboolean' from value of type 'gchararray'
  return Gio.Application.run(self, *args, **kwargs)
E:/newfolder/tartube-2.4.221-64bit-portable/mingw64/lib/python3.10/site-packages/gi/overrides/Gio.py
:42: Warning: unable to set property 'style-set' of type 'gboolean' from value of type 'gchararray'
hararray'
  Gtk.TreeStore.set(self, treeiter, columns, values)
E:/newfolder/tartube-2.4.221-64bit-portable/mingw64/lib/python3.10/site-packages/gi/overrides/Gtk.py:
1333: Warning: unable to set property 'underline-set' of type 'gboolean' from value of type 'gchararr
ay'
  Gtk.TreeStore.set(self, treeiter, columns, values)
E:/newfolder/tartube-2.4.221-64bit-portable/mingw64/lib/python3.10/site-packages/gi/overrides/Gtk.py:
1333: Warning: unable to set property 'weight-set' of type 'gboolean' from value of type 'gchararray'

  Gtk.TreeStore.set(self, treeiter, columns, values)
E:/newfolder/tartube-2.4.221-64bit-portable/mingw64/lib/python3.10/site-packages/gi/overrides/Gtk.py:
1333: Warning: unable to set property 'style-set' of type 'gboolean' from value of type 'gchararray'
  Gtk.TreeStore.set(self, treeiter, columns, values)
E:/newfolder\tartube-2.4.221-64bit-portable\home\user\tartube\tartube/mainwin.py:9680: Warning: unabl
e to set property 'strikethrough-set' of type 'gboolean' from value of type 'gchararray'
  self.video_index_treestore.remove(tree_iter)
E:/newfolder\tartube-2.4.221-64bit-portable\home\user\tartube\tartube/mainwin.py:9680: Warning: unabl
e to set property 'underline-set' of type 'gboolean' from value of type 'gchararray'
  self.video_index_treestore.remove(tree_iter)
E:/newfolder\tartube-2.4.221-64bit-portable\home\user\tartube\tartube/mainwin.py:9680: Warning: unabl
e to set property 'weight-set' of type 'gboolean' from value of type 'gchararray'
  self.video_index_treestore.remove(tree_iter)
E:/newfolder\tartube-2.4.221-64bit-portable\home\user\tartube\tartube/mainwin.py:9680: Warning: unabl
e to set property 'style-set' of type 'gboolean' from value of type 'gchararray'
  self.video_index_treestore.remove(tree_iter)
```

and if you click 'Update yt-dlp', it seems already satisfied as below. of course, it doesn't work and popup same window.

```
Starting update operation, installing/updating yt-dlp
..\..\..\mingw64\bin\python3.exe ..\..\..\mingw64\bin\pip3-script.py install --upgrade --no-dependencies yt-dlp
Requirement already satisfied: yt-dlp in e:/newfolder/tartube-2.4.221-64bit-portable/mingw64/lib/python3.10/site-packages (2023.1.6)
Update operation finished
```

but if you move tartube folder to original path( E:\tartube-2.4.221-64bit-portable\\!backup ), it works again.
it seems something is missing.
Furthermore, it is portable so it should be able to work without specifying new path of any files.

> E:\newfolder\tartube-2.4.221-64bit-portable\downloadsAndDbFiles (X)

I would like to be able to manually change this path like below
> .\downloadsAndDbFiles (O)

**Comment by zjarhd** [comment]
original path is F:\storage_211202\tartube\
and I moved it to F:\newfolder\tartube\
tested with https://www.youtube.com/watch?v=gcsxAddFaoA



![스크린샷(50)](https://user-images.githubusercontent.com/37343691/213004334-902c77d7-814c-4b60-ac56-c973d314cae4.png)

![스크린샷(51)](https://user-images.githubusercontent.com/37343691/213004666-9c4897da-1ffb-41b3-8f3f-0dbfc0fde605.png)

It had worked on original path
![스크린샷(49)](https://user-images.githubusercontent.com/37343691/213004775-940e8226-df33-48e7-9795-36103ed8f093.png)

**Comment by axcore** [comment]
Great, that's the same error I can see.

**Comment by axcore** [comment]
Small update: I don't know yet how to fix this issue.

**Comment by axcore** [comment]
Fixed in v2.5.0. 

If you move the portable Tartube folder to a new location on your filesystem, you will need to reinstall yt-dlp. This can be done from Tartube's main menu. (It didn't work before, but now it does.)

Other components (such as FFmpeg) do not need to be reinstalled.

If you see the 'Fatal error in launcher: U' message, you should also see a Tartube message reminding you to update yt-dlp.
